compiler/ast/src/functions/input/function_input.rs

```diff
@@ -20,15 +20,32 @@ use leo_span::Span;
 use serde::{Deserialize, Serialize};
 use std::fmt;
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub enum ParamMode {
+    Const,
+    Private,
+    Public,
+}
+
+impl fmt::Display for ParamMode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use ParamMode::*;
+
+        match self {
+            Const => write!(f, "const"),
+            Private => write!(f, "private"),
+            Public => write!(f, "public"),
+        }
+    }
+}
+
 /// A function parameter.
 #[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct FunctionInputVariable {
     /// The name the parameter is accessible as in the function's body.
     pub identifier: Identifier,
     /// Is it a const parameter?
-    pub const_: bool,
-    /// Is it a mutable parameter?
-    pub mutable: bool,
+    pub mode: ParamMode,
```

Comment:
```suggestion
    mode: ParamMode,
```
Can this field be made private?
We need to make sure the mode cannot be accessed or changed after being set.
Instead we can use a public getter method for access.
```
pub fn get_mode(&self) -> ParamMode {
...
```

Reply:
I don't think we even pass the AST as mutable anywhere. More so, it's not like we are going to change this. So I don't see the point, but I can make the change.

Reply:
thank you